calidns is Linux-only

It's linux-only due to recvmmsg.

```
calidns.cc:37:33:   required from here
/usr/local/include/c++/4.9.2/ext/new_allocator.h:114:35: error: invalid application of 'sizeof' to incomplete type 'recvThread(const std::vector<Socket*>*)::mmsghdr'
       { return size_t(-1) / sizeof(_Tp); }
                                   ^
Makefile:2366: recipe for target 'calidns.o' failed
gmake[3]: *** [calidns.o] Error 1
gmake[3]: Leaving directory '/root/pdns/pdns'
Makefile:1923: recipe for target 'all' failed
gmake[2]: *** [all] Error 2
gmake[2]: Leaving directory '/root/pdns/pdns'
Makefile:545: recipe for target 'all-recursive' failed
gmake[1]: *** [all-recursive] Error 1
gmake[1]: Leaving directory '/root/pdns'
Makefile:477: recipe for target 'all' failed
gmake: *** [all] Error 2
```

--- a/configure.ac
+++ b/configure.ac
@@ -132,7 +132,9 @@ PDNS_CHECK_MKDOCS
 PDNS_CHECK_LINKCHECKER
 
 dnl Checks for library functions.
-AC_CHECK_FUNCS_ONCE([strcasestr localtime_r])
+AC_CHECK_FUNCS_ONCE([strcasestr localtime_r recvmmsg])
+
+AM_CONDITIONAL([HAVE_RECVMMSG], [test "x$ac_cv_func_recvmmsg" = "xyes"])
 
 AS_IF([test "x$lt_cv_dlopen" = "xno"],
   [AC_MSG_ERROR([Your system does not support dlopen])]

--- a/pdns/Makefile.am
+++ b/pdns/Makefile.am
@@ -73,7 +73,6 @@ bin_PROGRAMS = \
 
 if TOOLS
 bin_PROGRAMS += \
-	calidns\
 	dnsbulktest \
 	dnsgram \
 	dnsreplay \
@@ -86,6 +85,11 @@ bin_PROGRAMS += \
 	nproxy \
 	nsec3dig \
 	saxfr
+
+if HAVE_RECVMMSG
+bin_PROGRAMS += calidns
+endif
+
 endif
 
 EXTRA_PROGRAMS = \

